CHANGELOG.rst

```diff
@@ -12,6 +12,8 @@ Features
 - Moved connection related modules into new `connections` subpacket
 - Added the ability to repeat sending of packages within a given time or count.
 - Added optional timeout and threshold to quit infinite connection retries
+- Reworked Monitors, consolidated interface. Breaking change: session no longer has netmon_options and procmon_options.
+- `SessionInfo` has had attributes renamed; procmon_results and netmon_results are deprecated and now aliases for monitor_results and monitor_data respectively.
```

Comment:
Just pass the dict you previously supplied to procmon_options to the procmons set_options method with unpacking. You need to keep a reference though, so you have to create an instance and assign it to a local variable before instantiating Target or adding it to the monitor list.

Then, it's just `my_procmon.set_options(**procmon_options)`, or use keyword arguments if that's more to your liking. Dict unpacking in this way is nice; and I guess this is one of the reasons why I need to update the examples.

Reply:
OK, so Before:
```python
procmon = pedrpc.Client(procmon_host, procmon_port)
procmon_options = {}
procmon_options['start_commands'] = [procmon_start]
session = Session(
    target=Target(
        connection=SocketConnection(target_host, target_port, proto='tcp'),
        procmon=procmon,
        procmon_options=procmon_options,
    ),
)
```

After:
```python
procmon = pedrpc.Client(procmon_host, procmon_port)
procmon_options = {}
procmon_options['start_commands'] = [procmon_start]
procmon.set_options(**procmon_options)
session = Session(
    target=Target(
        connection=SocketConnection(target_host, target_port, proto='tcp'),
        procmon=procmon,
    ),
)
```
Like that?

Reply:
Kinda, however the set_options call can only be made once the monitor is alive, which is why I introduced the moitor_alive callbacks in the first place. When I come to think about it, that leads to kinda messy code however. I think the alive callback should be moved to the Monitors itself, so that the correct options are supplied there.

I don't really know what's the best way to do this. Setting of the options has to be delayed until a connection is made if the monitor RPCs. Maybe those Monitors should keep a shadow option dictionary that is updated everytime the RPC client reconnects? That'd be the user-friendliest way to solve the dilemma, I guess.

Reply:
I've decided that I'll go forward with the shadow-option-dict. I'll have a look how this is best implemented. This way, it also ensures that a known option state is present at each reconnect.

Reply:
@mistressofjellyfish OK. Could you post a new "After" sample to fix what I posted in the above reply? Just to get an idea of what this looks like for users. Thanks!

Reply:
With the new option cache code in place, your example should work now. I'll update the ftp_with_procmon.py example to reflect the new style, that should do it then, right?